com.unity.render-pipelines.core/Runtime/Debugging/ShaderDebugPrintManager.cs

```diff
@@ -0,0 +1,323 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine.Assertions;
+
+namespace UnityEngine.Rendering
+{
+    public sealed class ShaderDebugPrintManager
+    {
+        private static readonly ShaderDebugPrintManager instance = new ShaderDebugPrintManager();
+
+        private const int DebugUAVSlot = 7;
+        private const int FramesInFlight = 4;
+        private const int MaxBufferElements = 1024 * 16; // Must match the shader size definition
+
+        private List<GraphicsBuffer> m_OutputBuffers = new List<GraphicsBuffer>();
+
+        private List<Rendering.AsyncGPUReadbackRequest> m_readbackRequests =
+            new List<Rendering.AsyncGPUReadbackRequest>();
+
+        // Cache Action to avoid delegate allocation
+        private Action<AsyncGPUReadbackRequest> m_bufferReadCompleteAction;
+
+        private int m_FrameCounter = 0;
+        private bool m_FrameCleared = false;
+
+        private string m_OutputLine = "";
+        private Action<string> m_OutputAction;
+
+        private static readonly int m_ShaderPropertyIDInputMouse = Shader.PropertyToID("_ShaderDebugPrintInputMouse");
+        private static readonly int m_ShaderPropertyIDInputFrame = Shader.PropertyToID("_ShaderDebugPrintInputFrame");
+
+        enum DebugValueType
```

Comment:
I would add a comment that this enum matches the variables defined in `com.unity.render-pipelines.core/ShaderLibrary/ShaderDebugPrint.hlsl`